File: Models/Records/DestinyRecordComponent.cs
namespace bulkybook.Models;

public class DestinyRecordComponent
{
    public Int32 state {get;set;}
    public List<DestinyObjectiveProgress> objectives {get;set;}
    public List<DestinyObjectiveProgress> internalObjectives {get;set;}
    public Int32 intervalsRedeemedCount {get;set;}
    public Int32 completedCount {get;set;}
    public List<Boolean> rewardVisibilty {get;set;}
    
}

File: Scripts/Models/DestinyRecordComponent.ts
import { DestinyObjectiveProgress } from "./DestinyObjectiveProgress";
export interface DestinyRecordComponent
{
    state: number;
    objectives: Array<DestinyObjectiveProgress>;
    internalObjectives: Array<DestinyObjectiveProgress>;
    intervalsRedeemedCount: number;
    completedCount: number;y
    rewardVisibilty: Array<boolean>;
}